share/wake/lib/system/sources.wake

```diff
@@ -58,6 +58,67 @@ export def source (file: String): Result Path Error =
         x, Nil = Pass x
         _      = unreachable "a quoted RegExp can never match more than one distinct String"
 
+# Allows claiming of a file so long as the file is within
+# wakeroot. Keep in mind that if another job produces this
+# file, claim will fail. Likewise if you claim a file then
+# any job that produces this file will fail. Claim should
+# be thought of as a "virtual" job and wake demands that
+# all paths be produced by exactly one job, virtual or
+# otherwise.
+#
+# For source files please use `source`. For files outside
+# of wakeroot please use `claimFileAsPath` or `claimFileAsPathIn`.
+# `claim` is good for claiming files of previous builds that
+# are not being used again this build.
```

Comment:
I know this is literally the main reason it's being written, but I feel this might be a bit of a dangerous recommendation as part of the docs.  Maybe add a sentence making it clear that we're talking about files which provide an explicit re-entry point, rather than anything which might not be being rebuilt simply due to caching?
```suggestion
# are not being built again this build.
```